pkg/proxy/proxier.go

```diff
@@ -463,28 +464,29 @@ func (proxier *Proxier) OnUpdate(services []api.Service) {
 	glog.V(4).Infof("Received update notice: %+v", services)
 	activeServices := util.StringSet{}
 	for _, service := range services {
-		activeServices.Insert(service.Name)
-		info, exists := proxier.getServiceInfo(service.Name)
+		cachekey := types.NewNamespacedName(service.Namespace, service.Name).CacheKey()
```

Comment:
CacheKey() returns the NamespacedName object itself, not a string.  I originally wrote CacheKey() to make it easy to flip between using strings and objects (which I did a few times during testing and debugging).  If we've settled on using the object as the key, CacheKey() can be removed, I guess. I don't feel terribly strongly either way. Let me know your preference.